examples/tls-termination/haproxy/ingress-tls-default.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: app
+spec:
+  tls:
+  - hosts:
+    - foo.bar
+  rules:
+  - host: foo.bar
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: nginx
```

Comment:
That's a recurring theme with all controllers. They will mostly have the same setup, and maintainig these steps in N differnt docs makes it a lot easier to end up with an invalid doc because eg: the maintainer of controller-x didn't realize they need to upate the doc and everyone else did. 

I'm fine mentioning the same pre-requisites section as the other controller examples (https://github.com/kubernetes/ingress/tree/master/examples/tls-termination/nginx#prerequisites), and adding a little more at the end, eg
```
Prerequisites

You need a TLS cert and a test HTTP service for this example. 
You will also need to make sure you Ingress targets exactly one 
Ingress controller by specifying the ingress.class annotation, 
and that you have an ingress controller running in your cluster.

FYI, you can create the tls cert as follows
and the HTTP Service as follows
```